cns/hnsclient/hnsclient_windows.go

```diff
@@ -132,24 +170,416 @@ func createHnsNetwork(hnsNetwork *hcsshim.HNSNetwork) error {
 	}
 	hnsRequest := string(buffer)
 
-	// Create the HNS network.
-	log.Printf("[Azure CNS] HNSNetworkRequest POST request:%+v", hnsRequest)
-	hnsResponse, err := hcsshim.HNSNetworkRequest("POST", "", hnsRequest)
-	log.Printf("[Azure CNS] HNSNetworkRequest POST response:%+v err:%v.", hnsResponse, err)
+	// Create HNS network.
+	log.Printf("[Azure CNS] Creating HNS network: %+v", hnsRequest)
+	if _, err := hcsshim.HNSNetworkRequest("POST", "", hnsRequest); err != nil {
+		return fmt.Errorf("ERROR: Failed to create network due to error: %v", err)
+	}
 
-	return err
+	return nil
 }
 
 // deleteHnsNetwork calls HNS to delete the network with the provided name
 func deleteHnsNetwork(networkName string) error {
-	hnsNetwork, err := hcsshim.GetHNSNetworkByName(networkName)
-	if err == nil {
-		// Delete the HNS network.
-		var hnsResponse *hcsshim.HNSNetwork
-		log.Printf("[Azure CNS] HNSNetworkRequest DELETE id:%v", hnsNetwork.Id)
-		hnsResponse, err = hcsshim.HNSNetworkRequest("DELETE", hnsNetwork.Id, "")
-		log.Printf("[Azure CNS] HNSNetworkRequest DELETE response:%+v err:%v.", hnsResponse, err)
+	network, err := hcsshim.GetHNSNetworkByName(networkName)
+	if err != nil {
+		// If error is anything other than networkNotFound return error
+		if _, networkNotFound := err.(hcsshim.NetworkNotFoundError); !networkNotFound {
+			return fmt.Errorf("[Azure CNS] ERROR: Failed GetHNSNetworkByName due to error: %v", err)
+		}
+
+		log.Errorf("[Azure CNS] Network: %s not found for deletion", networkName)
+
+		return nil
+	}
+
+	// Delete HNS network.
+	log.Printf("[Azure CNS] Deleting HNS network: %+v", network)
+	if _, err = hcsshim.HNSNetworkRequest("DELETE", network.Id, ""); err != nil {
+		return fmt.Errorf("ERROR: Failed to delete network: %s due to error: %v",
+			networkName, err)
+	}
+
+	return nil
+}
+
+// CreateCompartment creates windows network compartment
+func CreateCompartment() (int, error) {
+	log.Printf("[Azure CNS] CreateCompartment")
+
+	var (
+		err           error
+		compartmentID int
+		stdout        string
+	)
+
+	if _, err = os.Stat(compartmentManagementBinary); err != nil {
+		log.Errorf("[Azure CNS] ERROR: Unable to find %s needed for compartment creation",
+			compartmentManagementBinary)
+		return compartmentID, fmt.Errorf("ERROR: Unable to create the compartment")
+	}
+
+	args := compartmentManagementBinary + "/operation create"
+	log.Printf("[Azure CNS] Creating compartment: %v", args)
+
+	if stdout, err = platform.ExecuteCommand(args); err != nil {
+		log.Errorf("[Azure CNS] ERROR: Failed to create compartment due to error: %v", err)
```

Comment:
combine err returned and log output in a var?

Reply:
This code is updated to use the existing method. The error message is returned using stderr